pkg/iac/terraform/state/backend/gs_reader.go

```diff
@@ -0,0 +1,61 @@
+package backend
+
+import (
+	"context"
+	"io"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"github.com/pkg/errors"
+)
+
+const BackendKeyGS = "gs"
+
+type GSBackend struct {
+	bucketName    string
+	path          string
+	reader        io.ReadCloser
+	storageClient *storage.Client
+}
+
+func NewGSReader(path string) (*GSBackend, error) {
+	storageClient, err := storage.NewClient(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	bucketPath := strings.Split(path, "/")
+	if len(bucketPath) < 2 {
+		return nil, errors.Errorf("Unable to parse Google Storage path: %s. Must be BUCKET_NAME/PATH/TO/OBJECT", path)
+	}
+	bucketName := bucketPath[0]
+	key := strings.Join(bucketPath[1:], "/")
+
+	return &GSBackend{
+		bucketName:    bucketName,
+		path:          key,
+		storageClient: storageClient,
+	}, nil
+}
+
+func (s *GSBackend) Read(p []byte) (int, error) {
+	if s.reader == nil {
+		ctx := context.Background()
+		rc, err := s.storageClient.Bucket(s.bucketName).Object(s.path).NewReader(ctx)
+		if err != nil {
+			return 0, errors.WithMessage(err, s.path)
```

Comment:
Shouldn't the error message be a little more explicit than just the path ? Or maybe you can just let the original error go ?

Reply:
I agree, if we forward the original error we got this 
![image](https://user-images.githubusercontent.com/6154987/145563280-911005b2-ec4e-4131-b49b-e2afc03d43c8.png)

I think this is clear enough WDYT ?

Reply:
:+1: the path is already displayed so I don't think we need to add anything